GetStream/stream-video-js
# feat: use the shared sfu call object in react native

# Known issues

* Video streams from android may not work all the time. As android sends vp9 codec support. This will be fixed by the backend later. Call between the web and android works though. In my test call between iOS and Android broke it. 
* ~~`joinWithCombinedStream` and `publishWithCombinedStream` are temporary methods added to the SDK until we work on splitting audio and video streams on react-native.~~
* back gestures break the video streaming screen
* call leaving doesn't work reliably all the time (better to close and restart the app for now)
* we don't fully use the shared state store yet. We await for the shared hooks repo to use it. For now only activeParticipants observable is used.

# How to test the react-native app

Connect your android device or ios device with USB to your machine. Don't use emulators as video streams are not sent from there.

You need Xcode installed for iOS and/or Android SDK for running Android. Details: https://reactnative.dev/docs/environment-setup

Note: Running on iOS device may be prompt you with signing problems (contact me if so)

- Run `yarn`, there is a manual patch that needs to be installed. 
- Run `yarn build:client` to build the js client
- Move to `cd packages/react-native-dogfood/`
- If using ios device. Run `npx pod-install`
- If using android device. Run `adb reverse tcp:8081 tcp:8081` to connect device.
- Run `npm install -g ios-deploy && npx react-native run-ios --device` for iOS and/or `yarn start android` to run the app on your device

Open a call in the web dogfood app. Enter the same call-id on the react native app. You should receive the video streams on both sides 🤞
